v2/internal/genericarmclient/generic_client.go

```diff
@@ -115,16 +118,26 @@ func (client *GenericClient) BeginCreateOrUpdateByID(ctx context.Context, resour
 
 func (client *GenericClient) createOrUpdateByID(
 	ctx context.Context,
-	resourceID string,
+	resourceID,
 	apiVersion string,
 	resource interface{}) (*http.Response, error) {
 
 	req, err := client.createOrUpdateByIDCreateRequest(ctx, resourceID, apiVersion, resource)
 	if err != nil {
 		return nil, err
 	}
+
+	requestStartTime := time.Now()
 	resp, err := client.pl.Do(req)
+
+	// Ignoring error here as resourceID would always be correct at this stage.
+	resourceType, _ := metrics.GetTypeFromResourceID(resourceID)
```

Comment:
Maybe panic? I've made similar assertions in comments a few times and have been encouraged to turn them into **panic** calls so that the assumption is verified.

Reply:
I think we don;t wanna go into the panic state if resourceID is not parsed properly. If there's something wrong with the resourceID, we'll get an error from ARM client anyway. We're just recording metrics here. Rather just thinking of it now, if resourceID is not parsed properly, we can just log the error and have the whole resourceID recorded in the metrics?

What do you think about above?